### PR TITLE
[Bug] Fix AttributeError in RuleCollection dupe check

### PR DESCRIPTION
**Pull request by brokensound77**
## Issues
Fixes bug found in #1743 

## Summary
```
 f"Rule ID {rule.id} for {rule.name} collides with rule {id_map.get(rule.id).get('name')}"
AttributeError: 'TOMLRule' object has no attribute 'get'
```

### CODE REVIEW

File: detection_rules/rule.py [hunk 1]
```diff
@@ -525,7 +525,7 @@ def to_api_format(self, include_version=True) -> dict:
 class TOMLRule:
     contents: TOMLRuleContents = field(hash=True)
     path: Optional[Path] = None
-    gh_pr: Any = field(hash=False, compare=False, default=None, repr=None)
+    gh_pr: Any = field(hash=False, compare=False, default=None, repr=False)
```
Comment (brokensound77) [inline detection_rules/rule.py:528]: unrelated fix

**Review by w0rk3r — APPROVED** [review]
LGTM
